community/io/src/test/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCacheWithRealFileSystemIT.java

```diff
@@ -19,15 +19,24 @@
  */
 package org.neo4j.io.pagecache.impl.muninn;
 
+import org.junit.After;
 import org.junit.Rule;
 
+import java.io.IOException;
+
 import org.neo4j.io.fs.DefaultFileSystemAbstraction;
-import org.neo4j.test.rule.TargetDirectory;
+import org.neo4j.test.rule.TestDirectory;
 
 public class MuninnPageCacheWithRealFileSystemIT extends MuninnPageCacheTest
 {
     @Rule
-    public TargetDirectory.TestDirectory directory = TargetDirectory.testDirForTest( getClass() );
+    public TestDirectory directory = TestDirectory.testDirectory();
+
+    @After
+    public void tearDown() throws IOException
+    {
+        directory.cleanup();
+    }
```

Comment:
Why are these cleanups needed? The `tearDown` method is called regardless of the outcome of the test, but we generally desire to keep the test data around if the test fails. Each test should also be working with their own dedicated directory, so there should be no risk of test isolation failures.